src/library/dlgtagfetcher.cpp

```diff
@@ -337,10 +337,20 @@ void DlgTagFetcher::apply() {
 
         QFileInfo trackFileInfo = QFileInfo(m_track->getLocation());
 
-        QString coverArtCopyFilePath =
-                trackFileInfo.absoluteFilePath().left(
-                        trackFileInfo.absoluteFilePath().lastIndexOf('.') + 1) +
-                "jpeg";
+        // This is the track file path without an extension.
+        // It is used to compose the Cover Art file path
+        // To prevent failure due to different directory slashes 
+        // QFileInfo::absoluteFilePath() is used.
+        // Everything after the latest '.' is deleted.
+        // eg trackFilePathWithoutSuffix: "/media/foo/Abba - Super Trouper."
+        QString trackFilePathWithoutSuffix = trackFileInfo.absoluteFilePath().left(
+                trackFileInfo.absoluteFilePath().lastIndexOf('.') + 1);
+
+        // ImageFileData::readFormatFrom returns the extension
+        // Extension of cover art will be the result returned from CoverArtArchive
+        // eg coverArtCopyFilePath: "/media/foo/Abba - Super Trouper.jpeg"
+        QString coverArtCopyFilePath = trackFilePathWithoutSuffix +
+                ImageFileData::readFormatFrom(m_fetchedCoverArtByteArrays);
```

Comment:
Thanks.
Though, instead of creating and intermediate QString, let's compose the path in one step:
```suggestion
        // Compose the Cover Art file path. Use the correct file extension
        // by checking from the fetched image bytes (disregard extension
        // from file link).
        QString coverArtCopyFilePath = trackFileInfo.absoluteFilePath().left(
                trackFileInfo.absoluteFilePath().lastIndexOf('.') + 1) +
                ImageFileData::readFormatFrom(m_fetchedCoverArtByteArrays);
```

Reply:
Done! Thank you! 👍